### PR TITLE
Added chest block, updated textures

### PR DESCRIPTION
**Pull request by jordanb84**
The chest block does not yet allow movement of items (I'll do that next), but otherwise it and the inventory block are functioning.

For blocks, I've passed the camera in the render function for blocks that may need to know the camera position/offset. For example, the chest needs the camera to unproject the mouse position when checking if it's tapped.

This also includes the updated textures. I've updated all textures I could find from http://opengameart.org/content/voxel-pack

Display 0

Display 1

### CODE REVIEW

**Comment by jordanb84** [comment]
Oh, I forgot a way to close the chest's inventory for mobile. There is a "Close" button already there for inventories, but it doesn't seem to function.

**Comment by jmrapp1** [comment]
I love the new block sprites you added! 

Looking at your code, the only thing I can foresee being categorized as an issue is that when the player opens a chest the world still updates. Whether this is an issue or a feature, I'm not sure. The argument I can see players making is that they can't see the world when in a chest so it may not be "fair" to let the world update. We'll see if this becomes an issue in the future and worry about it if it does, but otherwise great work!

File: core/src/com/jmrapp/terralegion/game/views/screen/InventoryScreen.java [hunk 1]
```diff
@@ -207,7 +207,6 @@ public void update() {
 	@Override
 	public void render(SpriteBatch sb) {
 		sb.setProjectionMatrix(camera.combined);
-		sb.begin();
```
Comment (ghost) [inline core/src/com/jmrapp/terralegion/game/views/screen/InventoryScreen.java:210]: Removing this line causes a crash:

```
08-03 10:59:52.789 2359-2402/com.jmr.physics.android E/AndroidRuntime: FATAL EXCEPTION: GLThread 29
                                                                       Process: com.jmr.physics.android, PID: 2359
                                                                       java.lang.IllegalStateException: SpriteBatch.begin must be called before draw.
                                                                           at com.badlogic.gdx.graphics.g2d.SpriteBatch.draw(SpriteBatch.java:507)
                                                                           at com.badlogic.gdx.graphics.g2d.SpriteBatch.draw(SpriteBatch.java:502)
                                                                           at com.jmrapp.terralegion.game.views.screen.InventoryScreen.render(InventoryScreen.java:210)
                                                                           at com.jmrapp.terralegion.PhysicsGame.render(PhysicsGame.java:136)
                                                                           at com.badlogic.gdx.backends.android.AndroidGraphics.onDrawFrame(AndroidGraphics.java:416)
                                                                           at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1553)
                                                                           at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1253)
```

**Comment by jmrapp1** [comment]
Yes you must start the spritebatch before rendering anything.

**Comment by ghost** [comment]
Could you push the fix?

**Comment by jordanb84** [comment]
Oh, I removed that in favor of the object rendering the inventory screen starting/stopping the SpriteBatch.

For example, the chest renders an inventory screen. If the inventory screen starts the SpriteBatch, that causes an issue because it's already started. The outside object would have to stop the SpriteBatch before rendering the inventory screen, but that can also cause issues because the SpriteBatch may or may not already be stopped.

As a fix, I'll add the start/stop to the area in PhysicsGame where the inventory screen is rendered.

File: core/src/com/jmrapp/terralegion/game/world/block/BlockType.java [hunk 2]
```diff
@@ -3,7 +3,7 @@
 public enum BlockType {
 
 	AIR(0), DIRT(1), GRASS(2), STONE(3), DIAMOND(4), TORCH(5), STONE_WALL(6), COAL(8), DIRT_WALL(9),
-    WOOD(10), LEAVES(11);
+    WOOD(10), LEAVES(11), WOOD_CHEST(12);
```
Comment (ghost) [inline core/src/com/jmrapp/terralegion/game/world/block/BlockType.java:6]: @jmrapp1 ID 12 is already taken, the next available ID is 14.

**Pull request merged**